lib/bsdlib/bsd_os.c

```diff
@@ -305,7 +305,7 @@ void bsd_os_errno_set(int err_code)
 		errno = EKEYREJECTED;
 		break;
 	default:
-		errno = EINVAL;
+		errno = err_code;
```

Comment:
We're mixing error-spaces here. If a numeric value of one of these untranslated nrf_errno codes matches a value of some Zephyr errno code, the app could not tell which one was originally returned.

I'd rather use logger here, all in all it's an information to the developer - application will not know what to do with NRF_* error code anyway.

Reply:
Logging is a good idea, but perhaps we should still return something else than `EINVAL`.

Reply:
Actually I think we should try to `assert`. There should always be a matching translation between a nRF `errno` and Zephyr's. If there isn,t the `bsd_os.c` needs to be updated. If we let an unhandled `errno` slip, it might create bugs down the road when application developers expect certain results from an operation but receive something else.

So in this order I would:
1) assert
2) log
3) return magic number

Reply:
I think it makes sense to do this in such order.